scripts/index.ts

```diff
@@ -13,22 +13,17 @@ if (appConfig.features.useTansaProofing) {
 
 let body = angular.element('body');
 
-function loadConfigs(callback) {
-    // can't use network helpers since they depend on angular being loaded
-    var oReq = new XMLHttpRequest();
-
-    oReq.addEventListener('load', function() {
-        const serverConfig = JSON.parse(this.responseText).config;
-
-        Object.assign(appConfig, serverConfig);
-        callback();
-    });
-    oReq.open('GET', appConfig.server.url + '/client_config');
-    oReq.send();
+function loadConfigs() {
+    return fetch(appConfig.server.url + '/client_config', {
+        method: 'GET',
+        mode: 'cors',
+    }).then((res) => res.json().then((json) => {
```

Comment:
you should check response status: `(res) => res.ok ? res.json() : Promise.reject(res)`

Reply:
the application won't start if there's an error here, but I can add error handling so it looks nice :smile: